cartography/intel/kubernetes/rbac.py

```diff
@@ -0,0 +1,439 @@
+                        all_api_groups.append(api_group_name)
+
+            # Extend resources
+            if rule.resources:
+                for resource in rule.resources:
+                    if resource not in all_resources:
+                        all_resources.append(resource)
+
+            # Extend verbs
+            if rule.verbs:
+                for verb in rule.verbs:
+                    if verb not in all_verbs:
+                        all_verbs.append(verb)
+
+        result.append(
+            {
+                "id": f"{role.metadata.namespace}/{role.metadata.name}",
+                "name": role.metadata.name,
+                "namespace": role.metadata.namespace,
+                "uid": role.metadata.uid,
+                "creation_timestamp": get_epoch(role.metadata.creation_timestamp),
+                "resource_version": role.metadata.resource_version,
+                "api_groups": all_api_groups,
+                "resources": all_resources,
+                "verbs": all_verbs,
+            }
+        )
+    return result
+
+
+def transform_role_bindings(role_bindings: List[V1RoleBinding]) -> List[Dict[str, Any]]:
+    """
+    Transform Kubernetes RoleBindings into a list of dictionaries.
+    """
+    result = []
+    for rb in role_bindings:
+        # Only process ServiceAccount subjects
+        service_account_subjects = [
+            subject
+            for subject in (rb.subjects or [])
+            if subject.kind == "ServiceAccount"
+        ]
+
+        for subject in service_account_subjects:
+            result.append(
+                {
+                    "id": f"{rb.metadata.namespace}/{rb.metadata.name}/{subject.namespace}/{subject.name}",  # same role binding can be used for different service accounts so need a unique id
+                    "name": rb.metadata.name,
+                    "namespace": rb.metadata.namespace,
+                    "uid": rb.metadata.uid,
+                    "creation_timestamp": get_epoch(rb.metadata.creation_timestamp),
+                    "resource_version": rb.metadata.resource_version,
+                    "role_name": rb.role_ref.name,
+                    "role_kind": rb.role_ref.kind,
+                    "subject_name": subject.name,
+                    "subject_namespace": subject.namespace,
+                    "subject_service_account_id": f"{subject.namespace}/{subject.name}",
+                    "role_id": f"{rb.metadata.namespace}/{rb.role_ref.name}",
+                }
+            )
+    return result
+
+
+def transform_cluster_roles(cluster_roles: List[V1ClusterRole]) -> List[Dict[str, Any]]:
+    """
+    Transform Kubernetes ClusterRoles into a list of dictionaries.
+    Flattens rules into separate api_groups, resources, and verbs lists.
+    """
+    result = []
+    for cluster_role in cluster_roles:
+        # Flatten all rules into combined lists
+        all_api_groups = []
+        all_resources = []
+        all_verbs = []
```

Comment:
```suggestion
        all_api_groups = set()
        all_resources = set()
        all_verbs = set()
```
I noticed the "if <blah> not in list, append" behavior below so I think you want sets here to prevent duplicates